lib/ui/recommendation.js

```diff
@@ -17,10 +16,10 @@ function Recommendation(opts) {
 
 Recommendation.prototype = {
   init: function() {
-    // Note: there is a race between XBL modifying the XUL DOM to insert our
-    // this.el, and the scripts being loaded and initialized. Rather than try
-    // to fire a 'ready' event / set app state from the XBL <constructor>,
-    // which is sometimes unreliable, we instead poll the XUL DOM for the el.
+    // Note: the XUL DOM isn't updated to insert the
+    // 'universal-search-recommendation' element until the popup is opened
+    // once. So, poll for the element until it's ready, then attach listeners
+    // to it.
```

Comment:
We could probably up the polling time to 50 or 75 ms, since it won't prevent queries and the round trip time will likely take at least that long.